assert.Eventually() does not fail test when the timer expires

assert.Eventually() returns false (and not Fail(...)) when the timer expires.

As a result, the test does not fail.

So for example, including  `assert.Eventually(t, func()bool{return false}, time.Second, time.Millisecond))` in your test does not cause it to fail.  It will poll the function for a second, and then pass.

Is this intentional?  It seems surprising, since the other assert functions cause the test to fail when the condition is not met.

As I was seeing the same, I did some experimentation:

```golang
package main

import (
	"testing"
	"time"
	"github.com/stretchr/testify/assert"
	//	"github.com/stretchr/testify/require"
)

func TestEventually(t *testing.T) {

	assert.Eventually(t, func() bool {
		return false
	}, 10 * time.Second, 1 * time.Second, "always false")

}
```

With v1.4.0:

```console
duncan@linux-1cur:/space/git/testify-testcase> go test .
--- FAIL: TestEventually (10.00s)
    main_test.go:12:
                Error Trace:    main_test.go:12
                Error:          Condition never satisfied
                Test:           TestEventually
                Messages:       always false
FAIL
FAIL    github.com/dmacvicar/testify-testcase   10.004s
```

With v1.4.1-0.20191106224347-f1bd0923b832

```console
duncan@linux-1cur:/space/git/testify-testcase> go test .
go: finding github.com/stretchr/testify master
ok      github.com/dmacvicar/testify-testcase   10.005s
```

That can be traced to https://github.com/stretchr/testify/commit/f1bd0923b8320ddfb0a8a9d919b5aeb8b7a4cc40 which replaced a

```golang
return Fail(t, "Condition never satisfied", msgAndArgs...)
```

with

```golang
return false
```

--- a/assert/assertions.go
+++ b/assert/assertions.go
@@ -1579,7 +1579,7 @@ func Eventually(t TestingT, condition func() bool, waitFor time.Duration, tick t
 	for tick := ticker.C; ; {
 		select {
 		case <-timer.C:
-			return false
+			return Fail(t, "Condition never satisfied", msgAndArgs...)
 		case <-tick:
 			tick = nil
 			go func() { ch <- condition() }()

